### PR TITLE
OpenBMC rinv multiple options

### PR DESCRIPTION
**Pull request by gurevichmark**
#3889 

Allow multiple options for OpenBMC `rinv` command

```
[root@stratton01 xcat]# rinv f6u13 serial model cpu abc
f6u13: Error: Unsupported command: rinv abc

[root@stratton01 xcat]# rinv f6u13 serial
f6u13: SYSTEM SerialNumber : 1318DAA.........

[root@stratton01 xcat]# rinv f6u13 serial model
f6u13: SYSTEM Model : 8335-GTC........
f6u13: SYSTEM SerialNumber : 1318DAA.........

[root@stratton01 xcat]# rinv f6u13 serial model cpu
f6u13: SYSTEM Model : 8335-GTC........
f6u13: SYSTEM SerialNumber : 1318DAA.........
f6u13: CPU0 BuildDate : 1996-01-01 - 00:00:00
f6u13: CPU0 CORE0 Functional : 0
f6u13: CPU0 CORE0 Present : 0
f6u13: CPU0 CORE0 PrettyName :
f6u13: CPU0 CORE1 Functional : 0
f6u13: CPU0 CORE1 Present : 0
f6u13: CPU0 CORE1 PrettyName :
f6u13: CPU0 CORE2 Functional : 1
f6u13: CPU0 CORE2 Present : 1
f6u13: CPU0 CORE2 PrettyName :
f6u13: CPU0 CORE3 Functional : 1
f6u13: CPU0 CORE3 Present : 1
f6u13: CPU0 CORE3 PrettyName :
f6u13: CPU0 CORE4 Functional : 1
f6u13: CPU0 CORE4 Present : 1
f6u13: CPU0 CORE4 PrettyName :
f6u13: CPU0 CORE5 Functional : 1
f6u13: CPU0 CORE5 Present : 1
f6u13: CPU0 CORE5 PrettyName :
f6u13: CPU0 CORE6 Functional : 1
f6u13: CPU0 CORE6 Present : 1
f6u13: CPU0 CORE6 PrettyName :
f6u13: CPU0 CORE7 Functional : 1
f6u13: CPU0 CORE7 Present : 1
f6u13: CPU0 CORE7 PrettyName :
f6u13: CPU0 CORE8 Functional : 0
f6u13: CPU0 CORE8 Present : 0
f6u13: CPU0 CORE8 PrettyName :
f6u13: CPU0 CORE9 Functional : 0
f6u13: CPU0 CORE9 Present : 0
f6u13: CPU0 CORE9 PrettyName :
f6u13: CPU0 CORE10 Functional : 1
f6u13: CPU0 CORE10 Present : 1
f6u13: CPU0 CORE10 PrettyName :
f6u13: CPU0 CORE11 Functional : 1
f6u13: CPU0 CORE11 Present : 1
f6u13: CPU0 CORE11 PrettyName :
f6u13: CPU0 CORE12 Functional : 1
f6u13: CPU0 CORE12 Present : 1
f6u13: CPU0 CORE12 PrettyName :
f6u13: CPU0 CORE13 Functional : 1
f6u13: CPU0 CORE13 Present : 1
f6u13: CPU0 CORE13 PrettyName :
f6u13: CPU0 CORE14 Functional : 0
f6u13: CPU0 CORE14 Present : 0
f6u13: CPU0 CORE14 PrettyName :
f6u13: CPU0 CORE15 Functional : 0
f6u13: CPU0 CORE15 Present : 0
f6u13: CPU0 CORE15 PrettyName :
f6u13: CPU0 CORE16 Functional : 1
f6u13: CPU0 CORE16 Present : 1
f6u13: CPU0 CORE16 PrettyName :
f6u13: CPU0 CORE17 Functional : 1
f6u13: CPU0 CORE17 Present : 1
f6u13: CPU0 CORE17 PrettyName :
f6u13: CPU0 CORE18 Functional : 1
f6u13: CPU0 CORE18 Present : 1
f6u13: CPU0 CORE18 PrettyName :
f6u13: CPU0 CORE19 Functional : 1
f6u13: CPU0 CORE19 Present : 1
f6u13: CPU0 CORE19 PrettyName :
f6u13: CPU0 CORE20 Functional : 1
f6u13: CPU0 CORE20 Present : 1
f6u13: CPU0 CORE20 PrettyName :
f6u13: CPU0 CORE21 Functional : 1
f6u13: CPU0 CORE21 Present : 1
f6u13: CPU0 CORE21 PrettyName :
f6u13: CPU0 CORE22 Functional : 1
f6u13: CPU0 CORE22 Present : 1
f6u13: CPU0 CORE22 PrettyName :
f6u13: CPU0 CORE23 Functional : 1
f6u13: CPU0 CORE23 Present : 1
f6u13: CPU0 CORE23 PrettyName :
f6u13: CPU0 Cached : 0
f6u13: CPU0 FieldReplaceable : 1
f6u13: CPU0 Functional : 1
f6u13: CPU0 Manufacturer : IBM
f6u13: CPU0 Model :
f6u13: CPU0 PartNumber : 02AA766
f6u13: CPU0 Present : 1
f6u13: CPU0 PrettyName : PROCESSOR MODULE
f6u13: CPU0 SerialNumber : YA1934017890
f6u13: CPU0 Version : 20
f6u13: CPU1 BuildDate : 1996-01-01 - 00:00:00
f6u13: CPU1 CORE0 Functional : 0
f6u13: CPU1 CORE0 Present : 0
f6u13: CPU1 CORE0 PrettyName :
f6u13: CPU1 CORE1 Functional : 0
f6u13: CPU1 CORE1 Present : 0
f6u13: CPU1 CORE1 PrettyName :
f6u13: CPU1 CORE2 Functional : 1
f6u13: CPU1 CORE2 Present : 1
f6u13: CPU1 CORE2 PrettyName :
f6u13: CPU1 CORE3 Functional : 1
f6u13: CPU1 CORE3 Present : 1
f6u13: CPU1 CORE3 PrettyName :
f6u13: CPU1 CORE4 Functional : 1
f6u13: CPU1 CORE4 Present : 1
f6u13: CPU1 CORE4 PrettyName :
f6u13: CPU1 CORE5 Functional : 1
f6u13: CPU1 CORE5 Present : 1
f6u13: CPU1 CORE5 PrettyName :
f6u13: CPU1 CORE6 Functional : 1
f6u13: CPU1 CORE6 Present : 1
f6u13: CPU1 CORE6 PrettyName :
f6u13: CPU1 CORE7 Functional : 1
f6u13: CPU1 CORE7 Present : 1
f6u13: CPU1 CORE7 PrettyName :
f6u13: CPU1 CORE8 Functional : 1
f6u13: CPU1 CORE8 Present : 1
f6u13: CPU1 CORE8 PrettyName :
f6u13: CPU1 CORE9 Functional : 1
f6u13: CPU1 CORE9 Present : 1
f6u13: CPU1 CORE9 PrettyName :
f6u13: CPU1 CORE10 Functional : 1
f6u13: CPU1 CORE10 Present : 1
f6u13: CPU1 CORE10 PrettyName :
f6u13: CPU1 CORE11 Functional : 1
f6u13: CPU1 CORE11 Present : 1
f6u13: CPU1 CORE11 PrettyName :
f6u13: CPU1 CORE12 Functional : 0
f6u13: CPU1 CORE12 Present : 0
f6u13: CPU1 CORE12 PrettyName :
f6u13: CPU1 CORE13 Functional : 0
f6u13: CPU1 CORE13 Present : 0
f6u13: CPU1 CORE13 PrettyName :
f6u13: CPU1 CORE14 Functional : 1
f6u13: CPU1 CORE14 Present : 1
f6u13: CPU1 CORE14 PrettyName :
f6u13: CPU1 CORE15 Functional : 1
f6u13: CPU1 CORE15 Present : 1
f6u13: CPU1 CORE15 PrettyName :
f6u13: CPU1 CORE16 Functional : 1
f6u13: CPU1 CORE16 Present : 1
f6u13: CPU1 CORE16 PrettyName :
f6u13: CPU1 CORE17 Functional : 1
f6u13: CPU1 CORE17 Present : 1
f6u13: CPU1 CORE17 PrettyName :
f6u13: CPU1 CORE18 Functional : 1
f6u13: CPU1 CORE18 Present : 1
f6u13: CPU1 CORE18 PrettyName :
f6u13: CPU1 CORE19 Functional : 1
f6u13: CPU1 CORE19 Present : 1
f6u13: CPU1 CORE19 PrettyName :
f6u13: CPU1 CORE20 Functional : 1
f6u13: CPU1 CORE20 Present : 1
f6u13: CPU1 CORE20 PrettyName :
f6u13: CPU1 CORE21 Functional : 1
f6u13: CPU1 CORE21 Present : 1
f6u13: CPU1 CORE21 PrettyName :
f6u13: CPU1 CORE22 Functional : 0
f6u13: CPU1 CORE22 Present : 0
f6u13: CPU1 CORE22 PrettyName :
f6u13: CPU1 CORE23 Functional : 0
f6u13: CPU1 CORE23 Present : 0
f6u13: CPU1 CORE23 PrettyName :
f6u13: CPU1 Cached : 0
f6u13: CPU1 FieldReplaceable : 1
f6u13: CPU1 Functional : 1
f6u13: CPU1 Manufacturer : IBM
f6u13: CPU1 Model :
f6u13: CPU1 PartNumber : 02AA766
f6u13: CPU1 Present : 1
f6u13: CPU1 PrettyName : PROCESSOR MODULE
f6u13: CPU1 SerialNumber : YA1934017921
f6u13: CPU1 Version : 20
[root@stratton01 xcat]#
```

### CODE REVIEW

File: xCAT-openbmc-py/lib/python/agent/xcatagent/openbmc.py [hunk 1]
```diff
@@ -223,9 +223,16 @@ def rinv(self, nodesinfo, args):
...
+        # of that, we need to specify multiple identical choices. If only
+        # one optional choice is specified - [all|cpu|dimm|firm|model|serial]
+        # only one option at a time is allowed.
+        # If specified - [all][cpu][dimm][firm][model][serial], then multiple
+        # options are accepted, but they are required to be ordered,
+        # e.g "cpu dimm" will work, but not "dimm cpu"
         rinv_usage = """
         Usage:
-            rinv [-V|--verbose] [all|cpu|dimm|firm|model|serial]
+            rinv [-V|--verbose] [all|cpu|dimm|firm|model|serial] [all|cpu|dimm|firm|model|serial] [all|cpu|dimm|firm|model|serial] [all|cpu|dimm|firm|model|serial] [all|cpu|dimm|firm|model|serial]
```
Comment (xuweibj) [inline xCAT-openbmc-py/lib/python/agent/xcatagent/openbmc.py]: Hi @gurevichmark , I just have a question that if command is like ``rinv all all``, is it OK?

Comment (gurevichmark) [reply xCAT-openbmc-py/lib/python/agent/xcatagent/openbmc.py]: @xuweibj When you run something like `rinv cpu cpu` or `rinv all all`, no error is displayed, but only one REST API call is made to get the data and only one set of output is displayed.

**Comment by gurevichmark** [comment]
@whowutwut With this PR and #5185, the improvements are seen when doing `rinv` for `serial` or `model`.
Where as before we were calling `/xyz/openbmc_project/inventory/enumerate` to get all the data and filter out to display what was requested, now we call `/xyz/openbmc_project/inventory/system`. The time difference is 4 sec to call `/xyz/openbmc_project/inventory/system` vs. 10 sec to call `/xyz/openbmc_project/inventory/enumerate`.
The is no performance difference when doing `rinv` for `cpu` or `dimm`. To get that data we now call `/xyz/openbmc_project/inventory/system/chassis/motherboard/enumerate` which takes 10 sec, similar to the old `/xyz/openbmc_project/inventory/enumerate` call.

**Comment by gurevichmark** [comment]
@xuweibj If no additional questions, can you please merge.

**Comment by xuweibj** [comment]
Looks good to me. Thanks @gurevichmark .

**Pull request merged**